client/core/types.go

```diff
@@ -403,20 +403,28 @@ func coreOrderFromTrade(ord order.Order, metaData *db.OrderMetaData) *Order {
 	return corder
 }
 
+// needed to get recent matches on refresh
+type recentMatch struct {
+	Rate uint64
+	Qty  uint64
+	Age  uint64
+}
+
 // Market is market info.
 type Market struct {
-	Name            string        `json:"name"`
-	BaseID          uint32        `json:"baseid"`
-	BaseSymbol      string        `json:"basesymbol"`
-	QuoteID         uint32        `json:"quoteid"`
-	QuoteSymbol     string        `json:"quotesymbol"`
-	LotSize         uint64        `json:"lotsize"`
-	RateStep        uint64        `json:"ratestep"`
-	EpochLen        uint64        `json:"epochlen"`
-	StartEpoch      uint64        `json:"startepoch"`
-	MarketBuyBuffer float64       `json:"buybuffer"`
-	Orders          []*Order      `json:"orders"`
-	SpotPrice       *msgjson.Spot `json:"spot"`
+	Name            string         `json:"name"`
+	BaseID          uint32         `json:"baseid"`
+	BaseSymbol      string         `json:"basesymbol"`
+	QuoteID         uint32         `json:"quoteid"`
+	QuoteSymbol     string         `json:"quotesymbol"`
+	LotSize         uint64         `json:"lotsize"`
+	RateStep        uint64         `json:"ratestep"`
+	EpochLen        uint64         `json:"epochlen"`
+	StartEpoch      uint64         `json:"startepoch"`
+	MarketBuyBuffer float64        `json:"buybuffer"`
+	Orders          []*Order       `json:"orders"`
+	SpotPrice       *msgjson.Spot  `json:"spot"`
+	RecentMatches   []*recentMatch `json:"recentmatches"`
```

Comment:
Agree about that.  I don't think I was explicit this point in https://github.com/decred/dcrdex/pull/1663#discussion_r936023097, but this `Market` struct doesn't deal with order book things.  The Orders that are in this struct are your own.  The only possible violation there is the spot price, but we didn't include that in the OrderBook types.

Reply:
Also before we get too committed to the naming, `RecentMatches` and `RecentMatch" suggests that each entry is an actual match, but this is a distillation of an epoch's matches into a summary.  This is an epoch match summary.

Reply:
> The only possible violation there is the spot price, but we didn't include that in the OrderBook types.

We get the spot price from the `"price_feed"` subscription instead of the order book feed, and we expect to have it for all markets all of the time.

Reply:
Indeed!  I was thinking of candles.  My bad.